app/src/renderer/vuex/modules/connection.js

```diff
@@ -99,30 +115,26 @@ export default function({ node }) {
         }
       }, nodeHaltedTimeout) // default 30s
     },
-    nodeHasHalted({ commit }) {
-      console.log(`node has halted`)
-      clearTimeout(state.nodeHaltedTimeout)
-      state.nodeHaltedTimeout = undefined
-      commit(`setModalNodeHalted`, true)
-    },
-    pollRPCConnection({ state, dispatch }, timeout = config.block_timeout) {
-      if (state.nodeTimeout || state.stopConnecting) return
-
-      state.nodeTimeout = setTimeout(() => {
-        // clear timeout doesn't work
-        if (state.nodeTimeout && !state.mocked) {
-          state.connected = false
-          state.nodeTimeout = null
-          dispatch(`pollRPCConnection`)
-        }
+    async pollRPCConnection(
+      { state, dispatch, commit },
+      timeout = config.block_timeout
+    ) {
+      const { node } = state.externals
+      if (state.stopConnecting) return
+
+      try {
+        // TODO replace with ping when we implement ws connection ourselves
```

Comment:
Can you create an issue too ? 
```suggestion
        // TODO: replace with ping when we implement ws connection ourselves
```

Reply:
https://github.com/cosmos/voyager/issues/1858